flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/difference/FlowDifferenceDebouncer.java

```diff
@@ -0,0 +1,122 @@
+/*
+ *  Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com)
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.flowmodelgenerator.core.difference;
+
+import com.google.gson.JsonElement;
+
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Debouncing specifically designed for flow model difference requests. This debouncer ensures that difference
+ * processing is only executed after a specified delay has passed since the last invocation, cancelling any pending
+ * executions in between. This class follows the Singleton pattern, ensuring only one instance exists across the
+ * application for difference operations.
+ *
+ * @since 1.0.0
+ */
+public class FlowDifferenceDebouncer {
```

Comment:
Shouldn't we have a common class for debouncing any or most of the requests capturing for APIs?

Reply:
IMO, Even debouncing logic is different we have to limit the duplicating boilerplate debouncing implementations. It would be better if we can design a common implementation separately if we have time.

Reply:
I agree, but in this case, we need to explore the use cases for the debouncer and come up with a uniform design. Otherwise, the generic implementation will require frequent refactoring.